### PR TITLE
Take suggestions from flake8 simplify

### PR DESCRIPTION
**Pull request by wxtim**
This is a small change with no associated Issue.

The latest version of Flake8 simplify identified 3 locations where our code could be simpler.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Already covered by existing tests.
- [x] No change log entry required (change in code style not substance).
- [x] No documentation update required.

### CODE REVIEW

**Review by MetRonnie — APPROVED** [review]

File: cylc/flow/network/graphql.py [hunk 1]
```diff
@@ -66,10 +66,8 @@ def grow_tree(tree, path, leaves=None):
     """
     tree_loc = [tree, {}]
     for loc, key in enumerate(path):
-        if key in tree_loc[loc % 2]:
-            tree_loc[(loc + 1) % 2] = tree_loc[loc % 2][key]
-        else:
-            tree_loc[loc % 2][key] = tree_loc[(loc + 1) % 2]
+        tree_loc[(loc + 1) % 2] = tree_loc[loc % 2].get(
+            key, tree_loc[(loc + 1) % 2])
```
Comment (oliver-sanders) [inline cylc/flow/network/graphql.py:70]: Doesn't look like this does the same thing to me.

Comment (datamel) [reply cylc/flow/network/graphql.py:70]: I was just wondering the same.

Comment (wxtim) [reply cylc/flow/network/graphql.py:70]: It isn't.

Comment (datamel) [reply cylc/flow/network/graphql.py:70]: Maybe keep the original and add a skip?

Comment (wxtim) [reply cylc/flow/network/graphql.py]: Just pushed a fix up, alongside some tests for the unit in question.

File: tests/unit/network/test_graphql.py [hunk 2]
```diff
@@ -154,3 +157,32 @@ def test_null_setter(pre_result, expected_result):
     """Test the null setting of different data types/results."""
     post_result = null_setter(pre_result)
     assert post_result == expected_result
+
+
+@pytest.mark.parametrize(
```
Comment (MetRonnie) [inline tests/unit/network/test_graphql.py:162]: Wrong branch?

**Review by MetRonnie — APPROVED** [review]

File: cylc/flow/network/graphql.py (hunk 1 above)
Comment (MetRonnie) [reply cylc/flow/network/graphql.py]: I've opened an issue on flake8-simply: https://github.com/MartinThoma/flake8-simplify/issues/83

**Review by datamel — APPROVED** [review]
Nice to have the extra coverage, I have run these locally too. Thanks @wxtim, the change looks good to me.

**Comment by wxtim** [comment]
Merged because 2x review approval since I fixed comments from @oliver-sanders

**Pull request merged**